lib/logging/log.js

```diff
@@ -0,0 +1,415 @@
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * log.delete(function(err, apiResponse) {
+     *   if (!err) {
+     *     // The log was deleted.
+     *   }
+     * });
+     */
+    delete: true
+  };
+
+  this.name = name.replace(/^.*logs\//, '');
+  this.unformattedName = name;
+
+  ServiceObject.call(this, {
+    parent: logging,
+    baseUrl: '/logs',
+    id: this.name,
+    methods: methods
+  });
+}
+
+nodeutil.inherits(Log, ServiceObject);
+
+/**
+ * Return an array of log entries with the desired severity assigned.
+ *
+ * @private
+ *
+ * @param {object|object[]} entries - Log entries.
+ * @param {string} severity - The desired severity level.
+ */
+Log.assignSeverityToEntries_ = function(entries, severity) {
+  return arrify(entries).map(function(entry) {
+    return extend(true, {}, entry, {
+      metadata: {
+        severity: severity
+      }
+    });
+  });
+};
+
+/**
+ * Write a log entry with a severity of "ALERT".
+ *
+ * This is a simple wrapper around {module:logging/log#insert}. All arguments
+ * are the same as documented there.
+ *
+ * @example
+ * log.alert({
+ *   metadata: {
+ *     serviceName: 'compute.googleapis.com'
+ *   },
+ *   data: {
+ *     delegate: process.env.USER
+ *   }
+ * }, function(err, apiResponse) {});
+ */
+Log.prototype.alert = function(entry, labels, callback) {
+  this.insert(Log.assignSeverityToEntries_(entry, 'ALERT'), labels, callback);
+};
+
+/**
+ * Write log entries from a streaming object input.
+ *
+ * @example
+ * var logStream = log.createWriteStream();
+ *
+ * logStream.on('finish', function(err) {
+ *   // All entries have been written.
+ * });
+ *
+ * logStream.end({
+ *   // Log entry.
+ * });
+ */
+Log.prototype.createWriteStream = function() {
+  var self = this;
+  var writeStream = streamEvents(pumpify.obj());
+
+  writeStream.once('writing', function() {
+    var requestStream = self.request({
+      method: 'POST',
+      uri: '/entries:write'
+    });
+
+    requestStream.on('response', function(response) {
+      writeStream.emit('response', response);
+    });
+
+    function formatEntry_(entry, enc, next) {
+      next(null, self.formatEntry_(entry));
+    }
+
+    writeStream.setPipeline([
+      through.obj(formatEntry_),
+      JSONStream.stringify('{"entries":[', ',', ']}'),
+      requestStream
+    ]);
+  });
+
+  return writeStream;
+};
+
+/**
+ * Write a log entry with a severity of "CRITICAL".
+ *
+ * This is a simple wrapper around {module:logging/log#insert}. All arguments
+ * are the same as documented there.
+ *
+ * @example
+ * log.critical({
+ *   metadata: {
+ *     serviceName: 'compute.googleapis.com'
+ *   },
+ *   data: {
+ *     delegate: process.env.USER
+ *   }
+ * }, function(err, apiResponse) {});
+ */
+Log.prototype.critical = function(entry, labels, callback) {
+  var entries = Log.assignSeverityToEntries_(entry, 'CRITICAL');
```

Comment:
Same comment as in ".alert" wrapper.
Also, this code can be changed into 1 line like there :)